### PR TITLE
VIDSOL-599: Hide local screenshare view when sharing entire screen to prevent infinite mirror effect

### PR DESCRIPTION
**Pull request by ydumburs**
#### What is this PR doing?
Hides the local publisher screenshare preview when the user shares the entire screen (displaySurface==='monitor') to prevent infinite mirror effect. The behavior only applies to entire screen mode. Tab and Window sharing remain unchanged.

#### How should this be manually tested?
1. Start a meeting.
2. Share Entire Screen.
3. Verify that 
- The local preview is hidden.
- No infinite mirror effect appears.
- The remote participant can see the shared screen normally.
5. Stop sharing and confirm the UI returns to normal.
6. Share a Tab or Window and verify the local preview is still visible.
 
#### What are the relevant tickets?
A maintainer will add this ticket number.

Resolves [VIDSOL-599](https://jira.vonage.com/browse/VIDSOL-599)

#### Checklist
[x] Branch is based on `develop` (not `main`).
[ ] Resolves a `Known Issue`.
[ ] If yes, did you remove the item from the `docs/KNOWN_ISSUES.md`? 
[ ] Resolves an item reported in `Issues`.
If yes, which issue? [Issue Number](https://github.com/Vonage/vonage-video-react-app/issues/)?

### CODE REVIEW

File: frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx [hunk 1]
```diff
@@ -41,16 +41,20 @@ const ScreenSharePublisher = ({
...
+          box={box}
+          data-testid="screen-publisher-container"
+          hasVideo
+          ref={containerRef}
+          isScreenshare
+        >
+          <ScreenShareNameDisplay name={streamName} box={box} />
+        </VideoTile>
+
+        <div id="screen-share-hidden-message">You are sharing your screen.</div>
```
Comment (VZaphod) [inline frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx]: I think that we should extract the text to a localized String file

File: frontend/src/css/App.css [hunk 2]
```diff
@@ -151,8 +158,35 @@ html {
...
   font-size: 1rem;
 }
+
 @media (max-width: 375px) and (max-height: 667px) {
   .file-upload-drop-area-text {
     font-size: 0.9rem !important;
   }
 }
+
+body.screensharing-entire #screen-publisher-container {
```
Comment (johnny-quesada-developer) [inline frontend/src/css/App.css]: Thanks a lot for the changes.

We’re trying to use Tailwind exclusively and reduce the need for custom CSS as much as possible. Ideally, we would like to migrate this to Tailwind utility classes so the styles stay close to the component they affect and are easier to maintain.

That said, I see that you’re adding the class directly to the DOM element, so utility classes may not be the cleanest approach here. However, adding arbitrary global CSS is not ideal either.

You could instead create **ScreenSharePublisher.module.scss** with something like:

```scss
.overlay {
  @apply fixed inset-0 flex items-center justify-center
         text-xl font-medium
         bg-black/60 z-50 text-white
         pointer-events-none;
}
```

This would give you scoped CSS that will never collide with other components in the app.

File: frontend/src/hooks/useScreenShare.tsx [hunk 3]
```diff
@@ -33,9 +33,22 @@ const useScreenShare = (): UseScreenShareType => {
   const [screenshareVideoElement, setScreenshareVideoElement] = useState<
     HTMLVideoElement | HTMLObjectElement
   >();
+  const [isSharingEntireScreen, setIsSharingEntireScreen] = useState(false);
```
Comment (johnny-quesada-developer) [inline frontend/src/hooks/useScreenShare.tsx]: Hi @ydumburs ,

Since you are adding the css classes directly to the dom with `document.body.classList.add('screensharing-entire');`, you don't actually need **state** or side effects (**useEffect**)... You can easily 

```tsx
screenSharingPubRef.current?.on('videoElementCreated', (e) => {
          const videoEl = e.element as HTMLVideoElement;
          setScreenshareVideoElement(videoEl);
          const mediaStream = videoEl.srcObject as MediaStream | null;
          const track = mediaStream?.getVideoTracks?.()[0];
          const displaySurface = track?.getSettings?.()?.displaySurface;
          
          // adds the class
          document.body.classList.remove('overlay');
 });
```

and

```tsx
  const onScreenShareStopped = useCallback(() => {
             document.body.classList.remove('overlay');
```

No side effects, no re-renders, code is linear and easier to follow.

**Review by johnny-quesada-developer — CHANGES_REQUESTED** [review]

File: frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx (hunk 1 above)
Comment (ydumburs) [reply frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx]: Thanks for the suggestion. i moved the message text to the localization file.

File: frontend/src/css/App.css (hunk 2 above)
Comment (ydumburs) [reply frontend/src/css/App.css]: Thanks for the suggestion. I moved the styles to a new scss so the CSS is scoped to the component.

File: frontend/src/hooks/useScreenShare.tsx (hunk 3 above)
Comment (ydumburs) [reply frontend/src/hooks/useScreenShare.tsx]: I simplified the logic and removed the unnecessary state, updating it to the class directly as suggested.

**Comment by ydumburs** [comment]
Thanks for the suggestions. I've pushed the updates addressing the comments.

**Comment by OscarFava** [comment]
Pd: tests are failing, maybe you need to execute yarn test:integration update ?

File: frontend/src/hooks/useScreenShare.tsx [hunk 4]
```diff
@@ -33,28 +33,33 @@ const useScreenShare = (): UseScreenShareType => {
   const [screenshareVideoElement, setScreenshareVideoElement] = useState<
     HTMLVideoElement | HTMLObjectElement
   >();
-  const [isSharingEntireScreen, setIsSharingEntireScreen] = useState(false);
+  screenSharingPubRef.current?.on('videoElementCreated', (e) => {
```
Comment (Hossein-Movahed) [inline frontend/src/hooks/useScreenShare.tsx]: The `videoElementCreated` listener is registered twice, once outside `toggleShareScreen` (around line 36) and again inside it (line 100). The outer one re-attaches a new listener on every re-render, which can cause memory leaks and double firing. Since the inner one inside `toggleShareScreen` is sufficient, the outer registration can be removed entirely.

File: frontend/src/hooks/useScreenShare.tsx [hunk 5]
```diff
@@ -33,28 +33,33 @@ const useScreenShare = (): UseScreenShareType => {
...
       document.body.classList.add('screensharing-entire');
     } else {
       document.body.classList.remove('screensharing-entire');
     }
-    return () => {
-      document.body.classList.remove('screensharing-entire');
-    };
-  }, [isSharingEntireScreen]);
+
+    console.log('displaySurface:', displaySurface);
```
Comment (Hossein-Movahed) [inline frontend/src/hooks/useScreenShare.tsx]: Please remove the `console.log('displaySurface:', displaySurface)`

File: frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx [hunk 6]
```diff
@@ -53,7 +56,7 @@ const ScreenSharePublisher = ({
           <ScreenShareNameDisplay name={streamName} box={box} />
         </VideoTile>
 
-        <div id="screen-share-hidden-message">You are sharing your screen.</div>
+        <div className={styles.overlay}>{t('screenSharing.dialog.hiddenMessage')}</div>
```
Comment (Hossein-Movahed) [inline frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx]: As Johnny mentioned, we can handle this fully with Tailwind, no need for the SCSS module or the body.screensharing-entire #screen-publisher-container rule in App.css and the [ScreenSharePublisher.module.scss](https://github.com/Vonage/vonage-video-react-app/pull/381/changes/3e0dde06ceca88c0221b2f9b5d40b4b21bb296d8#diff-e9984a341f7b9757780317f6b7fc5338b244f8c3a30061b7b7287d647570b986) could be remove.

For example :
```
<div className="fixed inset-0 flex items-center justify-center text-[1.6rem] font-medium bg-black/60 z-[999] text-white pointer-events-none">
  {t('screenSharing.dialog.hiddenMessage')}
</div>```

**Review by Hossein-Movahed — COMMENTED** [review]
Hey, great work on this! A few things to address.

**Review by Hossein-Movahed — CHANGES_REQUESTED** [review]

File: frontend/src/hooks/useScreenShare.tsx (hunk 4 above)
Comment (ydumburs) [reply frontend/src/hooks/useScreenShare.tsx]: Thanks for pointing that out. I removed the duplicate listener and kept the one inside toggleShareScreen.

File: frontend/src/hooks/useScreenShare.tsx (hunk 5 above)
Comment (ydumburs) [reply frontend/src/hooks/useScreenShare.tsx]: My bad, I've removed the debug console log.

File: frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx (hunk 6 above)
Comment (ydumburs) [reply frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx]: Thanks for the suggestion, I've removed the SCSS module and the App.css rules and implemented the overlay using Tailwind as suggested. I'll submit the changes shortly.

**Comment by ydumburs** [comment]
Thanks for the suggestion. I've updated the logic as recommended. The update has been pushed.

File: frontend/src/locales/en.json [hunk 7]
```diff
@@ -174,6 +174,7 @@
   "screenSharing.dialog.action": "Start sharing your screen",
   "screenSharing.dialog.content": "Looks like there is someone else sharing their screen. If you continue, their screen is no longer going to be shared.",
   "screenSharing.dialog.title": "Do you want to share your screen?",
+  "screenSharing.dialog.hiddenMessage": "You are sharing your screen.",
```
Comment (Hossein-Movahed) [inline frontend/src/locales/en.json:185]: @ydumburs Please add `es`,`it`, es-mx` strings

File: frontend/src/hooks/useScreenShare.tsx [hunk 8]
```diff
@@ -1,4 +1,4 @@
-import { useState, useRef, useCallback } from 'react';
+import { useState, useRef, useCallback } from 'react'; //
 import { Publisher, initPublisher } from '@vonage/client-sdk-video';
```
Comment (Hossein-Movahed) [inline frontend/src/hooks/useScreenShare.tsx:2]: Please remove the `//`

File: frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx [hunk 9]
```diff
@@ -39,18 +41,42 @@ const ScreenSharePublisher = ({
     }
   }, [element, theme.shapes.borderRadiusLarge]);
   const streamName = publisher?.stream?.name ?? '';
+  const videoEl = element as HTMLVideoElement | undefined;
+  const mediaStream = videoEl?.srcObject as MediaStream | null;
+  const track = mediaStream?.getVideoTracks?.()[0];
+  const displaySurface = track?.getSettings?.()?.displaySurface;
```
Comment (Hossein-Movahed) [inline frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx]: `displaySurface` logic is duplicated
It's computed twice, once in `useScreenShare.tsx` (to set the body class) and again inside `ScreenSharePublisher.tsx` (to derive isEntireScreen). The clean fix is to expose `isEntireScreen` from `useScreenShare` and pass it as a prop to `ScreenSharePublisher`, removing the duplicated video track inspection entirely.

File: frontend/src/hooks/useScreenShare.tsx [hunk 10]
```diff
@@ -36,12 +36,14 @@ const useScreenShare = (): UseScreenShareType => {
 
   const onScreenShareStopped = useCallback(() => {
     setIsSharingScreen(false);
+    document.body.classList.remove('screensharing-entire');
```
Comment (Hossein-Movahed) [inline frontend/src/hooks/useScreenShare.tsx]: The `document.body.classList` calls are pointless and should be removed.

File: frontend/src/hooks/useScreenShare.tsx [hunk 11]
```diff
@@ -36,12 +36,14 @@ const useScreenShare = (): UseScreenShareType => {
...
     setScreenshareVideoElement(undefined);
     screenSharingPubRef.current = null;
   }, []);
 
   const unpublishScreenshare = useCallback(() => {
     if (screenSharingPubRef.current) {
+      document.body.classList.remove('screensharing-entire');
```
Comment (Hossein-Movahed) [inline frontend/src/hooks/useScreenShare.tsx]: Same here

File: frontend/src/hooks/useScreenShare.tsx [hunk 12]
```diff
@@ -80,7 +82,16 @@ const useScreenShare = (): UseScreenShareType => {
...
         screenSharingPubRef.current?.on('videoElementCreated', (e) => {
-          setScreenshareVideoElement(e.element);
+          const videoEl = e.element as HTMLVideoElement;
+          setScreenshareVideoElement(videoEl);
+          const mediaStream = videoEl.srcObject as MediaStream | null;
+          const track = mediaStream?.getVideoTracks?.()[0];
+          const displaySurface = track?.getSettings?.()?.displaySurface;
+          if (displaySurface === 'monitor') {
+            document.body.classList.add('screensharing-entire');
+          } else {
```
Comment (Hossein-Movahed) [inline frontend/src/hooks/useScreenShare.tsx]: Same here

File: frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx [hunk 13]
```diff
@@ -39,18 +41,42 @@ const ScreenSharePublisher = ({
...
+  const isEntireScreen = displaySurface === 'monitor';
   return (
```
Comment (Hossein-Movahed) [inline frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx:47]: A `absolute` and  `inset-0` fills the parent exactly,  no numbers `- 2` or `+4` or inline styles  needed. The `VideoTile` owns the positioning and the overlay just covers it.

The return could be:
```
return (
  box && (
    <VideoTile
      id="screen-publisher-container"
      box={box}
      data-testid="screen-publisher-container"
      hasVideo
      ref={containerRef}
      isScreenshare
    >
      {isEntireScreen ? (
        <div className="absolute inset-0 flex items-center justify-center text-xl font-medium bg-black text-white pointer-events-none rounded-2xl">
          {t('screenSharing.dialog.hiddenMessage')}
        </div>
      ) : (
        <ScreenShareNameDisplay name={streamName} box={box} />
      )}
    </VideoTile>
  )
);
```

**Review by Hossein-Movahed — CHANGES_REQUESTED** [review]
Thank you @ydumburs, just a few comments, and please add more tests, we need minimum 80% coverage for new code, and make sure the E2E test are passing

File: frontend/src/locales/en.json (hunk 7 above)
Comment (ydumburs) [reply frontend/src/locales/en.json:185]: Thanks, I added the message in other locale files.

File: frontend/src/hooks/useScreenShare.tsx (hunk 8 above)
Comment (ydumburs) [reply frontend/src/hooks/useScreenShare.tsx:2]: Removed //

File: frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx (hunk 9 above)
Comment (ydumburs) [reply frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx]: Thanks for the suggestion. I moved the displaySurface logic to useScreenShare and now expose isEntireScreen flag from the hook.

File: frontend/src/hooks/useScreenShare.tsx (hunk 10 above)
Comment (ydumburs) [reply frontend/src/hooks/useScreenShare.tsx]: Removed

File: frontend/src/hooks/useScreenShare.tsx (hunk 11 above)
Comment (ydumburs) [reply frontend/src/hooks/useScreenShare.tsx]: Removed

File: frontend/src/hooks/useScreenShare.tsx (hunk 12 above)
Comment (ydumburs) [reply frontend/src/hooks/useScreenShare.tsx]: Removed

File: frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx (hunk 13 above)
Comment (ydumburs) [reply frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx:47]: Thanks for the suggestion. Up updated the component so the overlay is rendered inside VideoTile using absolute inset-0, and removed the inline positioning styles. The overlay now simply covers the parent tile.

**Comment by ydumburs** [comment]
Hi @Hossein-Movahed @OscarFava, I can't access the SonarCloud details page from my account. Could you please share which files or lines are uncovered so I can add the missing tests?

**Comment by OscarFava** [comment]
> Hi @Hossein-Movahed @OscarFava, I can't access the SonarCloud details page from my account. Could you please share which files or lines are uncovered so I can add the missing tests?

frontend/src/hooks/useScreenShare.tsx

**Comment by Hossein-Movahed** [comment]
@ydumburs the E2E test is failing, the test now expects "**You are sharing your screen**." on the publisher side (Chrome always picks entire screen in E2E) instead of the **User One's screen**

```await expect(
      await pageOne.getByTestId('screen-publisher-container').getByText(`User One's screen`)
    ).toBeVisible();```

**Comment by ydumburs** [comment]
Thank you @Hossein-Movahed . Updated the screenshare test.

**Review by Hossein-Movahed — DISMISSED** [review]
LGTM

File: frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx [hunk 14]
```diff
@@ -49,7 +53,13 @@ const ScreenSharePublisher = ({
         ref={containerRef}
         isScreenshare
       >
-        <ScreenShareNameDisplay name={streamName} box={box} />
+        {isEntireScreen ? (
+          <div className="absolute inset-0 flex items-center justify-center text-xl font-medium bg-black text-white pointer-events-none">
```
Comment (OscarFava) [inline frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx]: We are not using our semantic tokens.

**Review by OscarFava — COMMENTED** [review]
Awesome work, minor comment.

File: frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx (hunk 14 above)
Comment (ydumburs) [reply frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx]: Thanks, I'll change those.

**Review by Hossein-Movahed — APPROVED** [review]

File: frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx [hunk 15]
```diff
@@ -49,7 +53,13 @@ const ScreenSharePublisher = ({
         ref={containerRef}
         isScreenshare
       >
-        <ScreenShareNameDisplay name={streamName} box={box} />
+        {isEntireScreen ? (
+          <div className="absolute inset-0 flex items-center justify-center text-xl font-medium bg-vera-dark-background text-vera-on-background pointer-events-none">
```
Comment (OscarFava) [inline frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx]: text-xl is wrong. You can use text-vera-heading-4

Comment (OscarFava) [reply frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx]: Same with font-medium. You should use our tokens.

**Review by OscarFava — COMMENTED** [review]
Minor comments

File: frontend/src/hooks/useScreenShare.tsx [hunk 16]
```diff
@@ -80,7 +84,12 @@ const useScreenShare = (): UseScreenShareType => {
         });
 
         screenSharingPubRef.current?.on('videoElementCreated', (e) => {
-          setScreenshareVideoElement(e.element);
+          const videoEl = e.element as HTMLVideoElement;
+          setScreenshareVideoElement(videoEl);
+          const mediaStream = videoEl.srcObject as MediaStream | null;
+          const track = mediaStream?.getVideoTracks?.()[0];
+          const displaySurface = track?.getSettings?.()?.displaySurface;
+          setIsEntireScreen(displaySurface === 'monitor');
```
Comment (Hossein-Movahed) [inline frontend/src/hooks/useScreenShare.tsx]: The `displaySurface` detection doesn't work in Firefox since `getSettings()` exists but `displaySurface` is not part of Firefox's implementation, so `isEntireScreen` will always be false there, causing the mirror effect.

A reliable fix is to fall back to comparing the track dimensions against the screen dimensions when `displaySurface` is `undefined`:
 
```
const settings = track?.getSettings?.();
const displaySurface = settings?.displaySurface;
// Firefox does not support displaySurface — fall back to dimension comparison
const isMonitor =
  displaySurface === 'monitor' ||
  (!displaySurface && settings?.width === screen.width && settings?.height === screen.height);
setIsEntireScreen(isMonitor);
``` 

Don't forget to add a test for the Firefox fallback path (where `displaySurface` is `undefined` but dimensions match the screen).

**Review by Hossein-Movahed — CHANGES_REQUESTED** [review]

File: frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx (hunk 15 above)
Comment (ydumburs) [reply frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx]: Thanks for the suggestion. I'll update the text and font.

File: frontend/src/hooks/useScreenShare.tsx (hunk 16 above)
Comment (ydumburs) [reply frontend/src/hooks/useScreenShare.tsx]: Thanks for the suggestion. I'm trying to hide the mirror effect for full screen sharing while keeping the Firefox PiP icon visible. Since the PiP icon seems to be browser-controlled and tied to the video element, hiding the video also hides the icon.
Do we want to:
1) Fully hide the mirror and lose PiP, or 
2) Keep PiP visible and partially mask the screen?

What would you prefer from a UX perspective?

File: frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx (hunk 15 above)
Comment (OscarFava) [reply frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx]: Is this fixed? I do not see the changes.

Comment (ydumburs) [reply frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx]: I fixed this locally but still waiting for @Hossein-Movahed 's answer on the next thread which I may need to change this part as well.

File: frontend/src/hooks/useScreenShare.tsx (hunk 16 above)
Comment (ydumburs) [reply frontend/src/hooks/useScreenShare.tsx]: Hi @Hossein-Movahed , just following up on this. Could you check the comment and let me know your thoughts when you have time?

Comment (Hossein-Movahed) [reply frontend/src/hooks/useScreenShare.tsx]: I'd go with fully hiding the mirror. The mirror effect is a clear visual bug that affects all full-screen sharers, while PiP on a local screenshare preview is rarely used. The overlay already gives enough feedback. The trade off is worth it i guess

Comment (ydumburs) [reply frontend/src/hooks/useScreenShare.tsx]: Got it, thanks for the confirmation. I'll fully hide the video element.

File: frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx (hunk 15 above)
Comment (ydumburs) [reply frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx]: Hi, I've addressed the requested change.

File: frontend/src/hooks/useScreenShare.tsx (hunk 16 above)
Comment (ydumburs) [reply frontend/src/hooks/useScreenShare.tsx]: Hi, I've addressed the requested changed - fixed firefox fallback. Also, resolved lint and conflict issues. Could you please take another look?

**Comment by Hossein-Movahed** [comment]
vcr:deploy

**Review by Hossein-Movahed — COMMENTED** [review]
@ydumburs , we need the string local in `de` as well

**Review by Hossein-Movahed — DISMISSED** [review]

**Review by OscarFava — DISMISSED** [review]

File: frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx [hunk 17]
```diff
@@ -24,9 +26,11 @@ const ScreenSharePublisher = ({
   box,
   element,
   publisher,
+  isEntireScreen,
 }: ScreenSharePublisherProps): ReactElement | undefined => {
```
Comment (behei-vonage) [inline frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx:31]: Hi @ydumburs, 

first of all, great job on this feature! 
I was curious about it so decided to test it and it works great! 

just a small comment, could you make sure please that `isEntireScreen` is added to the JSdoc on/after line 22 so that the docs are consistent?

thanks!

File: frontend/src/components/MeetingRoom/VideoTileCanvas/VideoTileCanvas.tsx [hunk 18]
```diff
@@ -33,6 +34,7 @@ export type VideoTileCanvasProps = {
  */
 const VideoTileCanvas = ({
   isSharingScreen,
+  isEntireScreen,
```
Comment (behei-vonage) [inline frontend/src/components/MeetingRoom/VideoTileCanvas/VideoTileCanvas.tsx:43]: same note here for the jsdoc but this time it looks like it's missing all of the props in the jsdoc on lines 28 through 34. Would you mind adding `isEntireScreen` and also the other ones please since you're in the area?

File: frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx (hunk 17 above)
Comment (ydumburs) [reply frontend/src/components/MeetingRoom/ScreenSharePublisher/ScreenSharePublisher.tsx:31]: Hi @behei-vonage , thanks for the suggestion! I'll add it to JSDoc.

File: frontend/src/components/MeetingRoom/VideoTileCanvas/VideoTileCanvas.tsx (hunk 18 above)
Comment (ydumburs) [reply frontend/src/components/MeetingRoom/VideoTileCanvas/VideoTileCanvas.tsx:43]: Thanks for the suggestion! I've updated the JSDoc to include all props, including isEntireScreen.

**Review by Hossein-Movahed — APPROVED** [review]

**Review by OscarFava — APPROVED** [review]
LGTM

**Comment by OscarFava** [comment]
@rserebrennykov This is ready for QA

**Review by rserebrennykov — APPROVED** [review]
LGTM

**Review by johnny-quesada-developer — APPROVED** [review]

**Comment by erwanrofim** [comment]
Hi @behei-vonage  👋

We're currently using this app and our users have been experiencing the infinite mirror effect issue fixed by this PR. We're glad to see it merged into `develop` on April 9th!

However, we'd like to avoid deploying a development build to production. Could you share an approximate timeline for the next release that will include this fix?

Thanks a lot for your work on this!

**Comment by OscarFava** [comment]
> Hi @behei-vonage 👋
> 
> We're currently using this app and our users have been experiencing the infinite mirror effect issue fixed by this PR. We're glad to see it merged into `develop` on April 9th!
> 
> However, we'd like to avoid deploying a development build to production. Could you share an approximate timeline for the next release that will include this fix?
> 
> Thanks a lot for your work on this!

Hello, the next release is on 23th of June. Thanks!
